src/utilities/util_verify.c

```diff
@@ -60,61 +60,65 @@ int
-      dump_blocks = dump_layout = dump_tree_shape = dump_pages = read_corrupt = stable_timestamp =
-        strict = false;
-    config = dump_offsets = uri = NULL;
+    abort_on_error = dump_all_data = dump_key_data = false;
+    dump_offsets = uri = NULL;
+    session_impl = (WT_SESSION_IMPL *)session;
+
+    WT_RET(__wt_scr_alloc(session_impl, 0, &config));
 
     while ((ch = __wt_getopt(progname, argc, argv, "acd:kSstu?")) != EOF)
         switch (ch) {
         case 'a':
             abort_on_error = true;
             break;
         case 'c':
-            read_corrupt = true;
+            WT_ERR(__wt_buf_catfmt(session_impl, config, "read_corrupt,"));
             break;
         case 'd':
             if (strcmp(__wt_optarg, "dump_address") == 0)
-                dump_address = true;
+                WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_address,"));
             else if (strcmp(__wt_optarg, "dump_blocks") == 0)
-                dump_blocks = true;
+                WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_blocks,"));
             else if (strcmp(__wt_optarg, "dump_layout") == 0)
-                dump_layout = true;
+                WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_layout,"));
             else if (strcmp(__wt_optarg, "dump_tree_shape") == 0)
-                dump_tree_shape = true;
+                WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_tree_shape,"));
             else if (WT_PREFIX_MATCH(__wt_optarg, "dump_offsets=")) {
                 if (dump_offsets != NULL) {
                     fprintf(
                       stderr, "%s: only a single 'dump_offsets' argument supported\n", progname);
                     return (usage());
                 }
                 dump_offsets = __wt_optarg + strlen("dump_offsets=");
+                WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_blocks=[%s],", dump_offsets));
             } else if (strcmp(__wt_optarg, "dump_pages") == 0)
-                dump_pages = true;
+                WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_pages,"));
             else
                 return (usage());
             break;
         case 'k':
             dump_key_data = true;
+            WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_key_data,"));
             break;
         case 'S':
-            strict = true;
+            WT_ERR(__wt_buf_catfmt(session_impl, config, "strict,"));
             break;
         case 's':
-            stable_timestamp = true;
+            WT_ERR(__wt_buf_catfmt(session_impl, config, "stable_timestamp,"));
             break;
         case 't':
-            do_not_clear_txn_id = true;
+            WT_ERR(__wt_buf_catfmt(session_impl, config, "do_not_clear_txn_id,"));
             break;
         case 'u':
             dump_all_data = true;
+            WT_ERR(__wt_buf_catfmt(session_impl, config, "dump_all_data,"));
             break;
         case '?':
             usage();
```

Comment:
For consistency with other util functions and safety, can we do:
```
case '?':
    usage();
    return (0);
default:
    return (usage());
}
```

Reply:
Maybe I'm missing something but how is it different from what we have now?

Reply:
oh my bad - didn't have the full file expanded!